### PR TITLE
[fundamental][tracing] Dev setup for promptflow-tracing

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

This PR targets to set up fundamental for new promptflow package `promptflow-tracing`:

- Add new dev setup script: `python scripts/dev-setup/main.py`
- Bump version to 1.0.0.dev0

**dev setup script**

help message

image

output (progress)

image

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,5 +1,7 @@
 * @microsoft/prompt-flow-approvers
 
+/src/promptflow-tracing/ @microsoft/promptflow-tracing
```
Comment (wangchao1230) [inline .github/CODEOWNERS:3]: I think we don't have a team called promptflow-tracing?

File: scripts/building/promptflow-tracing/dev-setup.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import json
```
Comment (wangchao1230) [inline scripts/building/promptflow-tracing/dev-setup.py:5]: I we have multi files fo rdifferent project, we should able to share some utils?

File: .github/CODEOWNERS (hunk 1 above)
Comment (zhengfeiwang) [reply .github/CODEOWNERS:3]: actually, i created one: <https://github.com/orgs/microsoft/teams/promptflow-tracing/members>; please let me know if we need another name...

File: scripts/building/promptflow-tracing/dev-setup.py (hunk 2 above)
Comment (zhengfeiwang) [reply scripts/building/promptflow-tracing/dev-setup.py]: have updated the script to work more like common one.

File: .github/CODEOWNERS (hunk 1 above)
Comment (wangchao1230) [reply .github/CODEOWNERS:3]: mmm... l originally thinks we should keep using the executor and sdk one. We should have an offline sync with Heyi on this.

File: scripts/dev-setup/main.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    deps = []
+    # cwd is already changed to the package folder in the context
+    # so we can safely open pyproject.toml here
+    with open("pyproject.toml", mode="r", encoding="utf-8") as f:
+        lines = f.readlines()
+        collect_flag = False
+        for line in lines:
+            line = line.strip()
+            if line.startswith("["):
+                if "[tool.poetry.group.dev.dependencies]" in line or "[tool.poetry.group.test.dependencies]" in line:
```
Comment (wangchao1230) [inline scripts/dev-setup/main.py:43]: does poetry require user to do something like this?

File: scripts/dev-setup/main.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+import site
+import typing
+from dataclasses import dataclass
+from pathlib import Path
+
+from test_resources import REGISTERED_TEST_RESOURCES_FUNCTIONS
+from utils import REPO_ROOT_DIR, change_cwd, print_blue, run_cmd
+
+PROMPT_FLOW_PKGS = [
+    "promptflow-tracing",
```
Comment (wangchao1230) [inline scripts/dev-setup/main.py:16]: how do promptflow-tools do dev setup now?

Comment (zhengfeiwang) [reply scripts/dev-setup/main.py:16]: tools has separate dev setup for now: <https://github.com/microsoft/promptflow/blob/main/src/promptflow-tools/README.dev.md>, so this script does not support tool for now

Comment (zhengfeiwang) [reply scripts/dev-setup/main.py:16]: if needed, i can add tools setup here

File: scripts/dev-setup/main.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (zhengfeiwang) [reply scripts/dev-setup/main.py:3]: no, i prefer to exposing this after 1.7.0 release

File: scripts/dev-setup/main.py (hunk 3 above)
Comment (zhengfeiwang) [reply scripts/dev-setup/main.py:43]: if user has poetry installed, we can leverage command `poetry export` to extract these deps to a req txt, so we don't need to do this brute force way; also, if user only uses poetry (without something like conda), then dev setup is very easy with simply `poetry install`, in that way, might not need a dev setup script at all.

File: .github/CODEOWNERS (hunk 1 above)
Comment (zhengfeiwang) [reply .github/CODEOWNERS]: reverted related commit to not block this pr

**Review by crazygao — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**